sopel_reddit/__init__.py

```diff
@@ -157,18 +158,31 @@ def video_info(bot, trigger, match):
     return say_post_info(bot, trigger, submission_id, False, True)
 
 
+@plugin.url(share_url)
+@plugin.output_prefix(PLUGIN_OUTPUT_PREFIX)
+def share_info(bot, trigger):
+    redirect_response = requests.get(
+        trigger, allow_redirects=False, headers={"User-Agent": USER_AGENT}
+    )
+    if redirect_response.status_code != 301:
+        return
```

Comment:
I'd like to point out that this won't correctly handle the `http://` links allowed by the `share_url` pattern. Those get a 307 to their `https://` counterpart (still with the `/s/tracking_gibberish` part unresolved).

Probably not too bad to move some of this logic into a separate function that can be called recursively. (I remember you said earlier today that reddit blocked your IP during testing. Was this check originally added because of that?)

Reply:
The http->https redirects are 301s for me, but yeah, unhandled case.

The status_code check was mostly for "if this isn't a redirect, the rest of this definitely won't work". The IP ban is a 403.

Reply:
All urls shall become https :grin: